packages/identity/src/idp-identity.ts

```diff
@@ -0,0 +1,339 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0 */
+export * from "./cdk-internals";
+
+import {
+  IdentityPool,
+  IdentityPoolProps,
+  UserPoolAuthenticationProvider,
+} from "@aws-cdk/aws-cognito-identitypool-alpha";
+import { Duration, RemovalPolicy, Annotations } from "aws-cdk-lib";
+import {
+  UserPoolIdentityProviderAmazon,
+  UserPoolIdentityProviderApple,
+  UserPoolIdentityProviderFacebook,
+  UserPoolIdentityProviderGoogle,
+  UserPoolIdentityProviderOidc,
+  UserPoolIdentityProviderSaml,
+  PasswordPolicy,
+  UserPool,
+  UserPoolClient,
+  AccountRecovery,
+  CfnUserPool,
+  Mfa,
+  // UserPoolIdentityProvider,
+  IUserPoolIdentityProvider,
+  UserPoolClientIdentityProvider,
+  UserPoolProps,
+  CognitoDomainOptions,
+  CustomDomainOptions,
+  UserPoolDomain,
+  IUserPool,
+  UserPoolClientOptions,
+} from "aws-cdk-lib/aws-cognito";
+import { Construct } from "constructs";
+import {
+  UserPoolIdentityProviderAmazonProps,
+  UserPoolIdentityProviderAppleProps,
+  UserPoolIdentityProviderFacebookProps,
+  UserPoolIdentityProviderGoogleProps,
+  UserPoolIdentityProviderOidcProps,
+  UserPoolIdentityProviderSamlProps,
+} from "./cdk-internals";
+import { IdentityProviderName } from "./cdk-internals/identityProviders";
+
+export const defaultPasswordPolicy: PasswordPolicy = {
+  minLength: 8,
+  requireLowercase: true,
+  requireUppercase: true,
+  requireDigits: true,
+  requireSymbols: true,
+  tempPasswordValidity: Duration.days(3),
+};
+
+export interface IdentityProviderProps {
+  readonly [IdentityProviderName.AMAZON]?: UserPoolIdentityProviderAmazonProps;
+  readonly [IdentityProviderName.APPLE]?: UserPoolIdentityProviderAppleProps;
+  readonly [IdentityProviderName.FACEBOOK]?: UserPoolIdentityProviderFacebookProps;
+  readonly [IdentityProviderName.GOOGLE]?: UserPoolIdentityProviderGoogleProps;
+  readonly [IdentityProviderName.OIDC]?: UserPoolIdentityProviderOidcProps;
+  readonly [IdentityProviderName.SAML]?: UserPoolIdentityProviderSamlProps;
```

Comment:
While the social ones are singletons I think it's possible to add multiple oidc/saml identity providers - is that something we should support? :)